modules/tinymce/src/core/test/ts/browser/html/SanitizationTest.ts

```diff
@@ -22,6 +22,13 @@ describe('browser.tinymce.core.html.SanitizationTest', () => {
       mimeType: 'text/html'
     }));
 
+    it('TINY-11500: Sanitize with invalid attribute name in content', () => testHtmlSanitizer({
```

Comment:
This isn't testing the validation properly. You need to also pass in `valid_elements: '*[*]'` to the Schema int he `testHtmlSanitizer` otherwise it will just filter out anything that doesn't have valid names. If you do the above these invalid attributes are still there.
Also we should have a higher up test where it tests this with a editor instance as well. These should be similar tests for SVGs/MathML etc that you can look at.

- [ ] Add proper filtering in the parser likely in the attribute filter that the DOMPurify runs

Reply:
Added a fiddle here that shows what is expected to work: https://fiddle.tiny.cloud/sUlDxSfbZs/0
Initial parsing and inserting of contents with these invalid attributes should not be possible if we fix that then they are filtered away for all other operations in the editor.